### PR TITLE
Add email field to VIMUser object

### PR DESCRIPTION
**Pull request by antonklevets**
#### Ticket

[908](https://vimean.atlassian.net/browse/VCI-908)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

- Should allow get user email

#### Implementation Summary

- Add `email` field to VIMUser object

#### Reviewer Tips

N/A

#### How to Test

N/A

### CODE REVIEW

**Review by gennadiiivanov — APPROVED** [review]

File: Sources/Shared/Models/VIMUser.h
Comment (gennadiiivanov) [inline Sources/Shared/Models/VIMUser.h]: let's keep empty space to save the structure

**Comment by gennadiiivanov** [comment]
Let's wait for approval from @rpassis

**Comment by rpassis** [comment]
Please hold off on merging this. I'll approve once I clarified something offline.

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Skipping duplicate build file in Compile Sources build phase: /Users/distiller/VimeoNetworking/Sources/Shared/Requests/Request+Apple.swift (in target ‘VimeoNetworking-iOS’ from project ‘VimeoNetworking’)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/8d8babfb191327b759887f98e49cb6ee338bf4d5/Sources/Shared/Requests/Request+PublishToSocial.swift#L46">Sources/Shared/Requests/Request+PublishToSocial.swift#L46</a></strong>: expression implicitly coerced from ‘String?’ to ‘Any’  <br /><code>
                .title: $0.title,
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/8d8babfb191327b759887f98e49cb6ee338bf4d5/Sources/Shared/Requests/Request+PublishToSocial.swift#L47">Sources/Shared/Requests/Request+PublishToSocial.swift#L47</a></strong>: expression implicitly coerced from ‘String?’ to ‘Any’  <br /><code>
                .description: $0.description,
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/8d8babfb191327b759887f98e49cb6ee338bf4d5/Sources/Shared/Requests/Request+PublishToSocial.swift#L79">Sources/Shared/Requests/Request+PublishToSocial.swift#L79</a></strong>: expression implicitly coerced from ‘String?’ to ‘Any’  <br /><code>
                .categoryID: $0.categoryID
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 231 tests, with 0 failures (0 unexpected) in 1.032 (1.270) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1680)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by gennadiiivanov** [comment]
@rpassis take a look now please

File: Sources/Shared/Models/VIMUser.m
Comment (rpassis) [inline Sources/Shared/Models/VIMUser.m]:
```suggestion
```

**Review by rpassis — CHANGES_REQUESTED** [review]
Hi there, apologies but it turns out my suggestion to use the `emails` field was not a good one. That field is used for email addresses other than the primary one so let's just revert this latest change.

File: Sources/Shared/Models/VIMUser.m
Comment (gennadiiivanov) [reply Sources/Shared/Models/VIMUser.m]: 👍

**Comment by gennadiiivanov** [comment]
@antonklevets let's try 2 different options in our repository:
1. Preferred: subclass User object and add email property to it.
2. Add private category for user object that includes "email" property.

**Comment by antonklevets** [comment]
> @antonklevets let's try 2 different options in our repository:
> 
> 1. Preferred: subclass User object and add email property to it.
> 2. Add private category for user object that includes "email" property.

We can't.
1. Subclass don't help me, because framework parse `VIMUser` object
2. Category don't support variables

**Comment by rpassis** [comment]
Hey @antonklevets can you point me to where/how you are making a request that returns a `VIMUser`? We may be able to get around that but I need more details on how you are obtaining the user that you currently need an email attribute for.

**Comment by antonklevets** [comment]
> Hey @antonklevets can you point me to where/how you are making a request that returns a `VIMUser`? We may be able to get around that but I need more details on how you are obtaining the user that you currently need an email attribute for.

Hey @rpassis I get `email` when call `getMeRequest` method and after login/create account, i.e. `vimeoClient.currentAccount?.user?.email`

**Comment by antonklevets** [comment]
@rpassis Don't you like `email` public property? I can make private property and get `email` via `keyValueDictionary` in our extension  if you ok

**Review by rpassis — APPROVED** [review]

**Pull request merged**
